src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,90 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double MAX_CAPACITY = 0.75;
+    private static final int INDEX_OF_CAPACITY = 2;
+    private int size;
+    private Node<K, V>[] elements;
+
+    {
+        elements = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        resizeIfNeed();
+        int currentCapacity = elements.length;
+
+        int hash = (key == null) ? 0 : Math.abs(key.hashCode() % elements.length);
```

Comment:
The hash calculation should be consistent across methods. Consider extracting the hash calculation logic into a separate method to avoid code duplication and ensure consistency.